src/shared/bootspec.c

```diff
@@ -1378,13 +1378,131 @@ static void boot_entry_file_list(
                 *ret_status = status;
 }
 
+static void copy_cmdline(
+                char *cmdline,
+                char **ret) {
+        char *temp = *ret;
+        *ret = strnappend(temp, cmdline, strlen(cmdline));
+        if (temp)
+                free(temp);
+        temp = *ret;
+        *ret = strnappend(temp, " ", 1);
+        if (temp)
+                free(temp);
+}
+
+static int print_addon(
+                BootEntryAddon *addon,
+                const char *addon_str) {
+        printf("  %s: %s\n", addon_str, addon->location);
+        printf("      cmdline: %s%s\n",special_glyph(SPECIAL_GLYPH_TREE_RIGHT),addon->cmdline);
+        return 0;
+}
+
+static int print_cmdline(
+                const BootEntry *e) {
+        _cleanup_free_ char *final_cmdline = NULL;
+        int r;
+
+        if (!strv_isempty(e->options)) {
+                _cleanup_free_ char *t = NULL, *t2 = NULL;
+                _cleanup_strv_free_ char **ts = NULL;
+
+                t = strv_join(e->options, " ");
+                if (!t)
+                        return log_oom();
+
+                ts = strv_split_newlines(t);
+                if (!ts)
+                        return log_oom();
+
+                t2 = strv_join(ts, "\n              ");
+                if (!t2)
+                        return log_oom();
+
+                printf("  defCmdline: %s\n", t2);
+                copy_cmdline(t2, &final_cmdline);
+        }
+
+        if (e->local_addon_list) {
+                LIST_FOREACH(addon_list, addon, e->local_addon_list) {
+                        /* Add space at the beginning of addon_str to align it correctly */
+                        r = print_addon(addon, " localAddon");
+                        if (r < 0)
+                                return r;
+                        copy_cmdline(addon->cmdline, &final_cmdline);
+                }
+        }
+
+        if (final_cmdline) {
+                printf(" finalCmdline: %s\n", final_cmdline);
+        }
+
+        return 0;
+}
+
+static int json_addon(
+                BootEntryAddon *addon,
+                const char *addon_str,
+                JsonVariant **array) {
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        int r;
+
+        r = json_variant_merge_objectb(
+                        &v, JSON_BUILD_OBJECT(
+                                JSON_BUILD_PAIR(addon_str, JSON_BUILD_STRING(addon->location)),
+                                JSON_BUILD_PAIR("cmdline", JSON_BUILD_STRING(addon->cmdline))));
+        if (r < 0)
+                return log_oom();
+
+        r = json_variant_append_array(array, v);
+        if (r < 0)
+                return log_oom();
+
+        return 0;
+}
+
+static int json_cmdline(
+                const BootEntry *e,
```

Comment:
no need for these line breaks

Reply:
not sure I understand what you mean

Reply:
the list of parameters is short, the function signature fits on a single line